### PR TITLE
Upgrade to datasworn 0.1.0 wip branch

### LINKED ISSUE #294: Upgrade to datasworn 0.1.0 wip branch

### PR DESCRIPTION
**Pull request by cwegrzyn**
- [x] Make code compile with 0.1.0
- [x] Deal with ID migration
- [x] Do we need to change the link syntax now that everything is prefixed?
  - [x] Implement name-based links for new handler
  - [x] Update docs for link syntax change
- [x] Curly brace template syntax change
- [x] present notice to user when homebrew file is incompatible
- [x] document the migration process

Fixes #294

### CODE REVIEW

**Comment by zkat** [comment]
What's the expected experience for folks who already have a bunch of journals with legacy links? Do you think we could automate that migration somehow?

**Comment by cwegrzyn** [comment]
> What's the expected experience for folks who already have a bunch of journals with legacy links? Do you think we could automate that migration somehow?

Yeah, that's the last thing on my list once I finish up the link handler rewrite.

Haven't decided yet whether I'll go the janky route or the correct route.

Janky: write a regex for old-style datasworn links in both raw and KDL-string-escaped forms, look up the link in the migration table included in Datasworn, and if a match is found, rewrite it to the new ID

Correct: use a markdown parser. Look for links and rewrite as above. Look for mechanics blocks, KDL parse them, recursively search for string args/properties, markdown parse them for links.

**Comment by zkat** [comment]
Just do the regex

also I’m not sure why you’re escaping `/` in KDL strings? That’s a safe character. I’ve been confused by that but kept forgetting to mention it

**Comment by cwegrzyn** [comment]
> also I’m not sure why you’re escaping `/` in KDL strings? That’s a safe character. I’ve been confused by that but kept forgetting to mention it

As far as I recall, I'm not! I've been thinking it's something that the kdljs library is doing. But also not sure why-- hadn't thought too hard about it, but now that you mention it, it _is_ the slash that is usually safe.

**Comment by cwegrzyn** [comment]
Huh

https://github.com/kdl-org/kdljs/blob/main/src/formatter.js#L19

**Comment by zkat** [comment]
Oh. Oh yeah we fixed that in v2. That’s a big oops. But yeah. I guess escaping is the right thing

**Comment by cwegrzyn** [comment]
Okay, special links done. That "just" leaves the migration piece.

File: src/link-handler.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        new MoveModal(plugin.app, plugin, entry.value).open();
+        break;
+      case "asset":
+        new AssetModal(plugin.app, plugin, entry.value).open();
+        break;
+      case "oracle":
+        new OracleModal(plugin.app, plugin, entry.value).open();
+        break;
+      default:
+        logger.warn("Clicked on link with '%s' but no modal exists!", entry.id);
```
Comment (zkat) [inline src/link-handler.ts:55]: Maybe a `Notice` here?

File: src/mechanics/node-builders.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export function createOracleNode(
 ): kdl.Node {
   return node("oracle", {
     properties: {
-      name: `[${name ?? oracleNameWithParents(roll.oracle)}](oracle:${roll.oracle.id})`,
+      name: `[${name ?? oracleNameWithParents(roll.oracle)}](${roll.oracle.id})`,
       // TODO: this is preposterous
```
Comment (zkat) [inline src/mechanics/node-builders.ts:84]: hahahaha I just noticed this

**Review by zkat — COMMENTED** [review]
this is some epic work. I'm so excited about it.

**Review by zkat — APPROVED** [review]

**Pull request merged**